test/functional/test_runner.py

```diff
@@ -285,11 +285,12 @@ def main():
 
     # Remove the test cases that the user has explicitly asked to exclude.
     if args.exclude:
-        exclude_tests = [re.sub("\.py$", "", test) + (".py" if ".py" not in test else "") for test in args.exclude.split(',')]
+        exclude_tests = [re.sub("\.py$", "", test) for test in args.exclude.split(',')]
         for exclude_test in exclude_tests:
-            if exclude_test in test_list:
-                test_list.remove(exclude_test)
-            else:
+            exclude_list = [test for test in test_list if test.split('.py')[0] == exclude_test]
```

Comment:
Could probably do with a comment here to explain why you're doing this (removing `<test_name>.py` and `<test_name>.py --arg` from the test list)